### DISCUSSION TITLE
JavaScript on Collection pages

### DISCUSSION THREAD

**Discussion by SimonXIX** [knowledgefutures/pubpub#1859] [Support]
We're looking at PubPub as one of the platforms in the Experimental Publishing strand of the COPIM project (https://www.copim.ac.uk/) and I've been playing around with customisation of Collection pages and landing pages. 

I've had some success with customisation through adding CSS in HTML blocks as discussed here (https://github.com/pubpub/pubpub/discussions/1307) but I was wondering if there's a similar way to inject small JavaScript elements into a page? Specifically I've been experimenting with a JavaScript snippet to randomly shuffle the divs containing links to a Collection's Pubs so that each visitor to the page would see the pubs in a different order. Any advice would be much appreciated.

**Comment by gabestein** [comment]
Unfortunately, we don't support custom Javascript because it would represent a large potential security risk for end users. But even if we did, I think randomly shuffling dom elements would cause problems with the rendering engine we use, React, which wants strict control over the dom. There might be a way to do it with flex-box that wouldn't change order elements, but even in that case I think you'd still need to add attributes to previously rendered elements, which I think would trigger a re-render and cause the same server/client mismatch issues (though I haven't tested it).

That said, I love the idea of a random ordering option, which we might be able to add to Pub blocks without much trouble, and which I can definitely see other folks taking advantage of. Do you think that would work for your purposes?

**Reply by SimonXIX** [reply]
Hi Gabe, thanks for that explanation. That makes perfect sense. 

Yeah, that kind of random ordering option would work perfectly for our use case. The idea is to give each reader a different route into the material: to represent them as random fragments to be picked up and read. So if you're able to add a random ordering option, that would be terrific for us.

**Reply by gabestein** [reply]
Cool! I added a ticket for it (https://github.com/pubpub/pubpub/issues/1863), and we'll discuss feasibility at next week's dev meeting.
